### PR TITLE
[Core]: Enable get_actor_name for actor runtime context

### LINKED ISSUE #35849: [Core] [FR] Access actor name (and other options) from within the actor
### Description

I wish to access the actor's metadata from within the actor (e.g. for logging purposes), and specifically the named actor's name.

### Use case

For example, for logging, I wish I could do:

```python
import ray
import ray.actor
import loguru

@ray.remote
class Actor:
    def __init__(self):
        actor_name = ray.actor.get_current_actor_options()['name']  # OR
        actor_name = ray.actor.get_current_actor_name()
        loguru.logger.add(sink=f'logs/{actor_name }.log')
```

or similar. Also, PID, worker ID, etc. and other metadata could be nice.

### PR DESCRIPTION
**Pull request by jonathan-anyscale**
## Why are these changes needed?
Expose current actor's name through `RuntimeContext()`

## Related issue number

Closes #35849

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/runtime_context.py [hunk 1]
```diff
@@ -253,13 +253,32 @@ def get_actor_id(self) -> Optional[str]:
...
+                     worker. Current mode: {self.worker.mode}"
             )
             return None
         actor_id = self.worker.actor_id
         return actor_id.hex() if not actor_id.is_nil() else None
 
+    def get_actor_name(self) -> Optional[str]:
+        """Get the current actor name in this worker.
+
+        Name of the actor of the current process.
```
Comment (rkooo567) [inline python/ray/runtime_context.py:266]:
```suggestion

```
This line seems duplicated

File: python/ray/runtime_context.py [hunk 2]
```diff
@@ -253,13 +253,32 @@ def get_actor_id(self) -> Optional[str]:
...
+        The name is in string format.
+
+        Returns:
+            The current actor name in this worker. None if there's no
+            actor name.
+        """
+        # only worker mode has actor_id
+        if self.worker.mode != ray._private.worker.WORKER_MODE:
+            logger.warning(
+                f"This method is only available when the process is a\
```
Comment (rkooo567) [inline python/ray/runtime_context.py:277]: same comment as above

**Review by rkooo567 — COMMENTED** [review]
~~Can you add a test?~~

File: python/ray/tests/test_runtime_context.py [hunk 3]
```diff
@@ -346,6 +346,19 @@ def foo(self):
     ray.get(actor.foo.remote())
 
 
+def test_actor_name(ray_start_regular):
```
Comment (rkooo567) [inline python/ray/tests/test_runtime_context.py:349]: nit can you just add this test in the previous test test_ids? Having a new suite will do ray.init which is pretty slow (and I think this test is too small, so it is probably better just piggybacking to the previous test)

Comment (jonathan-anyscale) [reply python/ray/tests/test_runtime_context.py:349]: Gotcha!

File: python/ray/tests/test_runtime_context.py [hunk 4]
```diff
@@ -345,6 +345,16 @@ def foo(self):
...
+    # actor name
+    @ray.remote
+    class NamedActor:
+        def name(self):
+            return ray.get_runtime_context().get_actor_name()
+
+    ACTOR_NAME = "actor_name"
+    named_actor = NamedActor.options(name=ACTOR_NAME).remote()
+    assert ray.get(named_actor.name.remote()) == ACTOR_NAME
+
```
Comment (jjyao) [inline python/ray/tests/test_runtime_context.py:357]: Can you also test the case where we don't specify `name=` and test what `get_actor_name()` returns.

**Review by jjyao — COMMENTED** [review]
Nice work!

File: python/ray/runtime_context.py [hunk 5]
```diff
@@ -254,12 +254,32 @@ def get_actor_id(self) -> Optional[str]:
...
 
+    def get_actor_name(self) -> Optional[str]:
+        """Get the current actor name of this worker.
+
+        This shouldn't be used in a driver process.
+        The name is in string format.
+
+        Returns:
+            The current actor name of this worker.
+            Return empty string if there's no actor name or if it's a driver
```
Comment (rkooo567) [inline python/ray/runtime_context.py:271]: This docstring is inconsistent. 

`If it is a driver == if it is not an actor`

but it says 

if it is a driver -> empty string
if it is not an actor -> None.

Can you update the docstring to

```
If a current worker is an actor, and if actor name doesn't exist, it returns an empty string.

If a current worker is not an actor, it returns None. 
```

File: python/ray/tests/test_runtime_context.py [hunk 6]
```diff
@@ -345,6 +345,20 @@ def foo(self):
...
+    # unnamed actor name
+    unnamed_actor = NamedActor.options().remote()
```
Comment (rkooo567) [inline python/ray/tests/test_runtime_context.py:359]: Can you add tests to verify all the specs from the API?

```
Make sure if it is a task, it returns "None".
Make sure if it is a driver, it returns "None".
Make sure if it has no actor name, it returns "".
Make sure if it is an actor and has a name, it returns the name
```

**Review by rkooo567 — APPROVED** [review]

**Pull request merged**